client/header/activity-panel/panels/inbox/action.js

```diff
@@ -37,30 +43,39 @@ class InboxNoteAction extends Component {
 			window.open( href, '_blank' );
 		}
 
-		this.setState( { inAction }, () =>
-			triggerNoteAction( noteId, action.id )
-		);
+		if ( dismissType ) {
```

Comment:
After testing: it seems like both the `all` and `this` do the same thing currently.  Is that expected?

Reply:
Ah, I see on the `add/4099_test_branch` branch that this gets updated.  Maybe we just check to see if `noteId` is set instead of passing an additional prop of `dismissType`.

Reply:
I like the idea. I made the change in the commit [462f4f9](https://github.com/woocommerce/woocommerce-admin/pull/4262/commits/462f4f9df7a6486b40df71b312feadcbe80eb6dc) (in the [PR 4262](https://github.com/woocommerce/woocommerce-admin/pull/4262) )